### DISCUSSION TITLE
Ability to change config in local .npmrc files

### DISCUSSION THREAD

**Discussion by DullReferenceException** [npm/feedback#419] [CLI] [answered]
I'm trying to automate some `.npmrc` updates, and to my surprise it doesn't seem possible to use `npm config set` to modify any `.npmrc` file other than the one in the home directory `~/.npmrc`. Even if the working directory is in a project folder with a `package.json` and `node_modules` directory, `npm config set` modifies the home directory config. I also don't see any command line options for `npm config set` to specify where it should write to.

Perhaps expecting the working directory to affect `npm config` doesn't make sense, so maybe supporting something like this would be preferable?

```shell
npm config --project set registry https://my.custom.registry.com/ 
```

...or even an environment variable would be something scriptable.

**Comment by DullReferenceException** [comment]
For what it's worth, there's at least a workaround, but it's counterintuitive. If you use an `npm_config_userconfig` environment variable, you can trick the CLI into writing a local file.

**Comment by MylesBorins** [comment, marked as answer]
There is an open RFC to add the ability to modify local .npmrc files!

https://github.com/npm/rfcs/pull/336
